clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -48,6 +53,16 @@
 public class ConsumerConfig extends AbstractConfig {
     private static final ConfigDef CONFIG;
 
+    // a list contains all the in-product assignor names. Should be updated when new assignor added.
+    // This is to help optimize the ConsumerCoordinator#performAssignment
+    public static final List<String> IN_PRODUCT_ASSIGNOR_NAMES =
+        Collections.unmodifiableList(Arrays.asList(
+            RANGE_ASSIGNOR_NAME,
+            ROUNDROBIN_ASSIGNOR_NAME,
+            STICKY_ASSIGNOR_NAME,
+            COOPERATIVE_STICKY_ASSIGNOR_NAME
```

Comment:
Question: Should we add `StreamsPartitionAssignor` here?

Reply:
No, I think it's correct to leave the Streams assignor out of this. Though I think it technically may not matter at the moment, since the Streams assignor will only assign topics from its own subscription and ignores the subscribed topics of the other members, we may want the flexibility to do something like this in the future.